spec/std/socket/udp_socket_spec.cr

```diff
@@ -52,6 +52,30 @@ describe UDPSocket do
       client.close
       server.close
     end
+
+    it "can join and transmit to multicast groups" do
+      udp = UDPSocket.new Socket::Family::INET
+      udp.bind "0.0.0.0", 2000
+
+      udp.multicast_loopback = false
+      udp.multicast_loopback?.should eq(false)
+      udp.multicast_loopback = true
+      udp.multicast_loopback?.should eq(true)
+
+      udp.multicast_hops = 4
+      udp.multicast_hops.should eq(4)
+      udp.multicast_hops = 1
+      udp.multicast_hops.should eq(1)
+
+      udp.multicast_interface Socket::IPAddress.new("0.0.0.0", 0)
+
+      addr = Socket::IPAddress.new("224.0.0.254", 2000)
+      udp.join_group(addr)
```

Comment:
should more than 1 addr be in the group to check that multicast can reach both of them?  also to distinguish from broadcast it might make sense to have an addr and check that it is not receiving anything.

Reply:
Broadcast isn't enabled and I don't think we need to complicate the test. We're not testing the OS level socket behaviour, we're testing crystals ability to configure a socket - which the test achieves.